src/GregClient/AuthProviders/IAuthProvider.cs

```diff
@@ -45,5 +45,11 @@ public interface IOAuth2AuthProvider : IAuthProvider
     public interface IOAuth2AccessTokenProvider
     {
         string GetAccessToken();
+    }
+
+    // Auth Provider Interface to get user account information.
+    public interface IAuthProviderUserAccount
```

Comment:
maybe IOAuth2UsernameProvider? to match the other naming scheme already here for other interfaces?